static/src/js/components/TemporalDisplay/__tests__/TemporalSelectionDropdown.test.js

```diff
@@ -1,147 +1,210 @@
+    const endTestObj = moment.utc('2015-01-01 12:00:00', true).format('YYYY-MM-DD HH:mm:ss')
+
+    fireEvent.change(inputs[1], { target: { value: endTestObj } })
+
+    expect(inputs[0].value).not.toBe(endTestObj)
+    expect(inputs[1].value).toBe(endTestObj)
   })
 
-  test('sets the state correctly with an invalid end date', () => {
-    const { enzymeWrapper } = setup()
-    const invalidDate = moment('2012-01-efss 12:00:00', 'YYYY-MM-DD HH:mm:ss', true)
-    const validStartDate = moment.utc('2012-01-01 12:00:00').toISOString()
-    const validEndDate = moment.utc('2012-01-02 12:00:00').toISOString()
+  test('sets the state correctly with an invalid start date', async () => {
+    const user = userEvent.setup()
+
+    setup()
+
+    await waitFor(async () => {
+      await user.click(screen.getByRole('button'))
+    })
+
+    const inputs = screen.getAllByRole('textbox')
+
+    const invalidDate = '2012-01-efss 12:00:00'
+    const validStartDate = moment.utc('2012-01-01 12:00:00').format('YYYY-MM-DD HH:mm:ss')
+    const validEndDate = moment.utc('2012-01-02 12:00:00')
 
-    enzymeWrapper.setState({ temporal: { startDate: validStartDate, endDate: validEndDate } })
-    enzymeWrapper.instance().setEndDate(invalidDate)
-    expect(enzymeWrapper.state().temporal.startDate).toBe('2012-01-01T12:00:00.000Z')
-    expect(enzymeWrapper.state().temporal.endDate).toBe('2012-01-efss 12:00:00')
+    try {
+      fireEvent.change(inputs[0], { target: { value: invalidDate } })
+    } catch (e) {
+      console.log(e)
+    }
+
+    expect(screen.getByText(/Invalid start date/i)).toBeInTheDocument()
+
+    fireEvent.change(inputs[0], { target: { value: validStartDate } })
+    fireEvent.change(inputs[1], { target: { value: validEndDate.toISOString() } })
+
+    expect(inputs[0].value).toBe(validStartDate)
+    expect(inputs[1].value).toBe(validEndDate.format('YYYY-MM-DD HH:mm:ss'))
   })
 
-  test('clears the values onClearClick', () => {
-    const { enzymeWrapper } = setup()
+  test('clears the values onClearClick', async () => {
     const onChangeQueryMock = jest.fn()
-    enzymeWrapper.setProps({
+    const user = userEvent.setup()
+
+    const validEndDate = '2019-03-30T00:00:00.000Z'
+    const validStartDate = '2019-03-29T00:00:00.000Z'
+
+    const { getByRole, getAllByRole } = setup({
       onChangeQuery: onChangeQueryMock,
       temporalSearch: {
-        endDate: '2019-03-30T00:00:00.000Z',
-        startDate: '2019-03-29T00:00:00.000Z'
+        endDate: validEndDate,
+        startDate: validStartDate
       }
     })
-    enzymeWrapper.setState({ open: true })
 
-    enzymeWrapper.instance().onClearClick()
+    await waitFor(async () => {
+      await user.click(getByRole('button'))
+    })
 
-    expect(onChangeQueryMock).toHaveBeenCalledWith({
-      collection: {
-        temporal: {}
-      }
+    const inputs = getAllByRole('textbox')
+    expect(inputs[0].value).toBe(moment.utc(validStartDate).format('YYYY-MM-DD HH:mm:ss'))
```

Comment:
I should've specified a little more clearly what I mean on that last this that last review. Do we need to call moment here?
```suggestion
    expect(inputs[0].value).toBe(validStartDate))
```
It seems like its already formatted and for react-testing-library I think we'd want to test against what is being rendered right on the DOM

Reply:
Same with the one on line 153

Reply:
@eudoroolivares2016 For line 153-4, that does need to get formatted to be as I wrote. The originals are formatted in ISO format, but on the DOM, they get converted to be 'YYYY-MM-DD HH:mm:ss'